app/models/comment.php

```diff
@@ -14,23 +14,120 @@ class Comment extends AppModel
         ),
     );
 
-    public static function getComments($offset, $limit, $id)
+    public static function getCommentsById($offset, $limit, $id)
     {
         $comments = array();
         $db = DB::conn();
         $rows = $db->rows("SELECT * FROM comment 
                         WHERE thread_id = ? ORDER BY created ASC LIMIT {$offset}, {$limit}", array($id));
-       
+
         foreach ($rows as $row) {
-           $comments[] = new Comment($row);
+            $favorites = array();
+            $user_favorites = $db->rows("SELECT * FROM favorites WHERE comment_id = ?", array($row['id']));
+            $total_favorites = Favorites::getCountByCommentId($row['id']);
+
+            foreach ($user_favorites as $v) {
+                $favorites[] = $v['user_id'];
+            }
+            $row['favorites'] = $favorites; 
+            $row['total_favorites'] = $total_favorites;
+            $comments[] = new Comment($row);
+        }  
+        return $comments;
+    }
+
+    public static function count($thread_id)
+    {
+        $db = DB::conn();
+        return (int) $db->value("SELECT COUNT(*) FROM comment WHERE thread_id = ?", array($thread_id));
+    }
+
+    public function edit($comment_id)
+    {
+        if (!$this->body) {
+            throw new RecordNotFoundException('Record Not Found');
+        }
+        
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $db->query('UPDATE comment SET body = ?, created = NOW()
+                    WHERE id = ?', array($this->body, $comment_id));
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+            throw $e;
+        }
+    }
+
+    public static function getCommentsByCommentId($comment_id)
+    {
+        $comments = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM comment WHERE id = ?", array($comment_id));
+
+        foreach($rows as $row) {
+            $comments[] = new self($row);
+        }
+
+        if (!$rows) {
+            throw new RecordNotFoundException('No Record found');
         }
         return $comments;
     }
 
-    public static function countComments($thread_id)
+    public static function getCommentsByUsername()
     {
+        $comments = array();
         $db = DB::conn();
-        return (int) $db->value("SELECT COUNT(*) FROM comment
-                            WHERE thread_id = ?", array($thread_id));
+        $rows = $db->rows("SELECT * FROM comment WHERE username = ?", array(Session::get('username')));
```

Comment:
please don't call `Session::get('username')` inside the model class. 
its the controller's duty. Controller class is the one responsible for the logic operation (get, parse or sanitise parameters).

I suggest you can do like this:
```php
Comment::getByUsername($username);
```